### DISCUSSION TITLE
Building via Julia with BinaryBuilder

### DISCUSSION THREAD

**Discussion by MKAbdElrahman** [NanoComp/meep#2014] [Q&A]
I really suffer each time I want  to build meep from source. Is it possible to automate this with  [BinaryBuilder.jl](https://github.com/JuliaPackaging/BinaryBuilder.jl).  We can build for any target machine and the process is so easy.

**Comment by smartalecH** [comment]
Sure, care to take an initial stab at this?

Note that the [conda builds](https://meep.readthedocs.io/en/latest/Installation/#conda-packages) are meant to solve this problem too (although we no longer have nightly builds, unfortunately).

**Reply by stevengj** [reply]
I use BinaryBuilder a lot for binary dependencies of Julia packages, but it is not really set up for Python dependencies, which pull in the whole Python ecosystem.  (e.g. https://github.com/JuliaPackaging/Yggdrasil doesn't have a binary package for numpy.)

Of course, we could build just the non-Python parts of Meep, but that wouldn't be useful to most people.  Since the API is centered on Python, it makes more sense as a conda package.
